packages/cw2-supported-interface/src/lib.rs

```diff
@@ -0,0 +1,130 @@
+We do provide a helper to construct a "raw query" to read the ContractInfo
+of any CW2-compliant contract.
+
+For more information on this specification, please check out the
+[README](https://github.com/CosmWasm/cw-plus/blob/main/packages/cw2/README.md).
+*/
+
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{CustomQuery, QuerierWrapper, QueryRequest, StdResult, Storage, WasmQuery};
+use cw_storage_plus::Item;
+
+pub const CONTRACT: Item<ContractVersion> = Item::new("contract_info");
+
+#[cw_serde]
+pub struct ContractVersion {
+    /// contract is the crate name of the implementing contract, eg. `crate:cw20-base`
+    /// we will use other prefixes for other languages, and their standard global namespacing
+    pub contract: String,
+    /// version is any string that this implementation knows. It may be simple counter "1", "2".
+    /// or semantic version on release tags "v0.7.0", or some custom feature flag list.
+    /// the only code that needs to understand the version parsing is code that knows how to
+    /// migrate from the given contract (and is tied to it's implementation somehow)
+    pub version: String,
+    /// supported_interface is an optional parameter returning a vector of string represents interfaces
+    /// that the contract support The string value is the interface crate names in Rust crate Registry.
+    /// This parameter is inspired by the EIP-165 from Ethereum.
+    /// Each string value should follow a common standard such as <Registry Domain>:<Crate Name>
+    /// e.g ["crates.io:cw721","crates.io:cw2"]
+    /// NOTE: this is just a hint for the caller to adapt on how to interact with this contract.
+    /// There is no guarantee that the contract actually implement these interfaces.
+    pub supported_interface: Option<Vec<String>>,
+}
+
+/// get_contract_version can be use in migrate to read the previous version of this contract
+pub fn get_contract_version(store: &dyn Storage) -> StdResult<ContractVersion> {
+    CONTRACT.load(store)
+}
+
+/// set_contract_version should be used in instantiate to store the original version, and after a successful
+/// migrate to update it
+pub fn set_contract_version<T: Into<String>, U: Into<String>>(
+    store: &mut dyn Storage,
+    name: T,
+    version: U,
+    supported_interface: Option<Vec<String>>,
+) -> StdResult<()> {
+    let val = ContractVersion {
+        contract: name.into(),
+        version: version.into(),
+        supported_interface,
+    };
+    CONTRACT.save(store, &val)
+}
+
+/// This will make a raw_query to another contract to determine the current version it
+/// claims to be. This should not be trusted, but could be used as a quick filter
+/// if the other contract exists and claims to be a cw20-base contract for example.
+/// (Note: you usually want to require *interfaces* not *implementations* of the
+/// contracts you compose with, so be careful of overuse)
+pub fn query_contract_info<T, CQ>(
```

Comment:
We could have two queries:
- query all available interfaces
- query if a specified interface is supported